http/src/handler.rs

```diff
@@ -319,11 +350,17 @@ impl<M: Metadata, S: Middleware<M>> RpcHandler<M, S> {
 		mut request: Vec<u8>,
 		uri: Option<hyper::Uri>,
 		metadata: M,
-	) -> Result<RpcPollState<M, S::Future>, hyper::Error> {
+	) -> Result<RpcPollState<M, S::Future>, BodyError> {
 		loop {
 			match body.poll()? {
-				// TODO [ToDr] reject too large requests?
 				Async::Ready(Some(chunk)) => {
+					// first check if this chunk itself is too large (in case it is already
+					// equal to `usize::MAX` where addition would overflow)
+					if chunk.len() > self.max_request_body_size
+						|| request.len() + chunk.len() > self.max_request_body_size
```

Comment:
It can still overflow if `max_request_body_size` is set to a large value?
Perhaps just use `checked_add` instead?

Reply:
True. I was assuming, no one would deviate too much from the 5 MiB default, but `checked_add` is surely the better approach.